### PR TITLE
docs(guide): convert reference/management/sdam-monitoring to rst

### PR DESCRIPTION
**Pull request by daprahamian**
Fixes NODE-2193

## Description

**What changed?**

I tried to update the structure of the SDAM notifications as best as possible. The problem is that `JSON.stringify` will not stringify `Map` and `Set`, so those show up as `{}`. If this is still adequate, then it's fine, but we might want to switch to something like ts interfaces?

**Are there any files to ignore?**

### CODE REVIEW

File: docs/guide/management/sdam-monitoring.txt [hunk 1]
```diff
@@ -1,3 +1,438 @@
 ===============
 SDAM Monitoring
 ===============
+
+The Node.js driver  features SDAM Monitoring events,
```
Comment (mbroadst) [inline docs/guide/management/sdam-monitoring.txt]: [nit]: weird extra space

File: docs/guide/management/sdam-monitoring.txt [hunk 2]
```diff
@@ -1,3 +1,438 @@
...
+       // does not serialize in JSON.stringify(). Check out the serverDescriptions
+       // from the serverDescriptionChanged event above to see the contents
+       "servers": {},
+       "stale": false,
+       "compatible": true,
+       "compatibilityError": null,
+       "logicalSessionTimeoutMinutes": 30,
+       "heartbeatFrequencyMS": 30000,
+       "localThresholdMS": 15,
+       "options": {
```
Comment (mbroadst) [inline docs/guide/management/sdam-monitoring.txt]: we should save `options` to the `TopologyDescription` as a non-enumerable property

File: docs/guide/management/sdam-monitoring.txt [hunk 3]
```diff
@@ -1,3 +1,438 @@
...
+   { durationMS: 20,
+     failure: new MongoError('some error'),
+     connectionId: 'localhost:27017' }
+
+serverOpening
+^^^^^^^^^^^^^
+
+.. code-block:: js
+
+   { topologyId: 0, name: 'localhost:27017' }
```
Comment (mbroadst) [inline docs/guide/management/sdam-monitoring.txt]: why not just put the named structure (the result of `console.dir`)

Comment (daprahamian) [reply docs/guide/management/sdam-monitoring.txt]: I wound up using a hybrid approach, where I did the console.dir, but them simplified some fields.

File: docs/guide/management/sdam-monitoring.txt (hunk 2 above)
Comment (daprahamian) [reply docs/guide/management/sdam-monitoring.txt]: okay, in the meantime, I'm just listing `options` as an `Object`

**Pull request merged**